GitExtUtils/GitUI/ThreadHelper.cs

```diff
@@ -16,11 +17,11 @@ public static class ThreadHelper
         private const int RPC_E_WRONG_THREAD = unchecked((int)0x8001010E);
 #pragma warning restore SA1139 // Use literal suffix notation instead of casting
 
-        private static JoinableTaskContext _joinableTaskContext;
-        private static JoinableTaskCollection _joinableTaskCollection;
-        private static JoinableTaskFactory _joinableTaskFactory;
+        private static JoinableTaskContext? _joinableTaskContext;
+        private static JoinableTaskCollection? _joinableTaskCollection;
+        private static JoinableTaskFactory? _joinableTaskFactory;
 
-        public static JoinableTaskContext JoinableTaskContext
+        public static JoinableTaskContext? JoinableTaskContext
```

Comment:
💭 It's a bit strange for this to be nullable but `JoinableTaskFactory` to not be. Do we ever check the return value of this for null? If not, consider changing the getter to:

```
return _joinableTaskContext ?? throw new InvalidOperationException("vs-threading is not configured")`
```

Since the setter will still need to _accept_ null, you can use one of the following approaches to indicate that the property can be assigned null but will never return it:

1. Declare the property as nullable, but apply `[NotNull]` to it
2. Declare the property as non-nullable, but apply `[AllowNull]` to it